src/math/matrix.c

```diff
@@ -51,12 +51,12 @@ int mat_multiply(struct mat_matrix_16b *a, struct mat_matrix_16b *b, struct mat_
 				x = a->data + a->columns * i;
 				y = b->data + j;
 				for (k = 0; k < b->rows; k++) {
-					s += (int32_t)(*x) * (*y);
-					x++;
+					/* Enhanced pointer arithmetic */
+					s += (int32_t)(*x++) * (*y);
 					y += y_inc;
 				}
-				*z = (int16_t)s; /* For Q16.0 */
-				z++;
+				/* Enhanced pointer arithmetic */
+				*z++ = (int16_t)s;
```

Comment:
I personally also like expressions like
```
*x++ = value;
```
but I don't think they actually improve readability, they're just more compact, so help scan the code faster. And I certainly don't expect these changes to affect performance. If you really want to convince me, you need to compare assembly before and after this change.